1.3.c

```diff
@@ -1,121 +1,111 @@
 #include <stdio.h>
+#include <string.h>
 #include <stdbool.h>
+// I am did't do maxsize , because my IDE cant scanf this.
 
-#define MAXIMUMSIZE 256
-
-bool isCorrectString(char* string)
+int IncomingDataStatus(char symbol, char state) 
 {
-    int state = 0;
-    char currentCharacter = -1;
-    int counter = 0;
-    bool isContinue = true;
-    while (isContinue && currentCharacter != '\0')
+    switch (symbol)
     {
-        currentCharacter = string[counter++];
-        switch (state)
+        case 1:
         {
-        case 0:
-            if ((currentCharacter >= 'A' && currentCharacter <= 'Z') || (currentCharacter >= 'a' && currentCharacter <= 'z'))
+            if (symbol >= 'A' && symbol <= 'Z' || symbol >= 'a' && symbol <= 'z')
             {
                 state = 1;
-                break;
             }

```

Comment:
```suggestion
            }
           break; 
```
В конце каждого случая должен быть либо break, либо return.
(-1)